lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+                </Tag>
+            </And>
+        </Filter>
+        <Status>Enabled</Status>
+        <NoncurrentVersionExpiration>
+            <NoncurrentDays>1</NoncurrentDays>
+        </NoncurrentVersionExpiration>
+        <AbortIncompleteMultipartUploads>
+            <DaysAfterInitiation>1</DaysAfterInitiation>
+        </AbortIncompleteMultipartUploads>
+    </Rule>
+</LifecycleConfiguration>
+ */
+
+/**
+ * Format of config:
+
+  config = {
+    rules = [
+        {
+            ruleID: <value>,
+            ruleStatus: <value>,
+            filter: {
+                rulePrefix: <value>,
+                tags: [
+                    {
+                        key: <value>,
+                        val: <value>
+                    },
+                    {
+                        key: <value>,
+                        val: <value>
+                    }
+                ]
+            },
+            actions: [
+                {
+                    actionName: <value>,
+                    days: <value>,
+                    date: <value>,
+                    deleteMarker: <value>
+                },
+                {
+                    actionName: <value>,
+                    days: <value>,
+                    date: <value>,
+                    deleteMarker: <value>,
+                },
+            ]
+        }
+    ]
+  };
+ */
+
+class LifecycleConfiguration {
+    /**
+     * Create a Lifecycle Configuration instance
+     * @param {string} xml - the parsed xml
+     * @return {object} - LifecycleConfiguration instance
+     */
+    constructor(xml) {
+        this._parsedXML = xml;
+        this._ruleIDs = [];
+        this._config = {};
+    }
+
+    /**
+     * Get the lifecycle configuration
+     * @return {object} - the lifecycle configuration
+     */
+    getLifecycleConfiguration() {
+        const rules = this._buildRulesArray();
+        if (rules.error) {
+            this._config.error = rules.error;
+        }
+        return this._config;
+    }
+
+    /**
+     * Build the this._config.rules array
+     * @return {object} - contains error if any rule returned an error
+     * or parsing failed
+     */
+    _buildRulesArray() {
+        const rules = {};
+        this._config.rules = [];
+        if (!this._parsedXML || this._parsedXML === '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml is undefined or empty');
+            return rules;
+        }
+        if (!this._parsedXML.LifecycleConfiguration &&
+            this._parsedXML.LifecycleConfiguration !== '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include LifecycleConfiguration');
+            return rules;
+        }
+        const lifecycleConf = this._parsedXML.LifecycleConfiguration;
+        const rulesArray = lifecycleConf.Rule;
+        if (!rulesArray || rulesArray.length === 0) {
```

Comment:
May be worthwhile to check `rulesArray` is an array before comparing its length.